File: Python files/CO2_loss_script_sal.py
#author: Riley Doyle
#date: 7/28/20
#file: CO2_loss_script_sal
#status: working 

#clear all
from IPython import get_ipython
get_ipython().magic('reset -sf')

from calc_Ks import *
from calc_alphas import *
from calc_CO2_loss import *
from calc_density import *
import numpy as np
import matplotlib.pyplot as plt
from calc_CO2_loss_sal import *

#figures with different alkalinities
S = 25
while S <= 45:
    T = 20 + 273.15
    Tc = 20; #celcius
    P = 10; #(dbar)
    t = Tc*1.00024;
    p = P/10;
    den = calc_density(S, t, p); #(kg/m3)
    K1 = calc_K1(T,S)*(den/1000) #mol/L
    pK1 = -np.log10(K1)
    K2 = calc_K2(T,S)*(den/1000) #mol/L
    pK2 = -np.log10(K2)
    Kh = calc_Kh(T,S)*(den/1000) #mol/L/atm
    PCO2 = 0.000416
    alkin = 2
    alkend = 27
    delalk = 5
    pHin = 6
    pHend = 8.2
    delpH = 0.1
    d = 0.15
    kLa = 0.5
    y = calc_CO2_loss(pK1, pK2, Kh, kLa, d, PCO2, pHin, pHend, delpH, alkin, alkend, delalk)
    plt.xlabel('pH')
    plt.ylabel('CO$_2$ loss to the atmosphere (g m$^{-2}$ day$^{-1})$')
    plt.legend(['alk = 2 meq/L', 'alk = 7 meq/L', 'alk = 12 meq/L', 'alk = 17 meq/L', 'alk = 22 meq/L'], frameon=False)
    plt.figure()
    S += 10

#figure with different salinities
Sin = 25
Send = 55
delS = 10
T = 20 + 273.15    
PCO2 = 0.000416
alk = 2.5
pHin = 6
pHend = 8.2
delpH = 0.1
d = 0.15
kLa = 0.5
y = calc_CO2_loss_sal (pK1, pK2, alk, d, PCO2, pHin, pHend, delpH, kLa, T, Sin, Send, delS)
plt.xlabel('pH')
plt.ylabel('CO$_2$ loss to the atmosphere (g m$^{-2}$ day$^{-1})$')
plt.legend(['S = 25 g/kg', 'S = 35 g/kg', 'S = 45 g/kg'], frameon=False)
plt.show()

File: Matlab files/CO2_loss_script_sal.m
%Author: Riley Doyle
%Date: 7/31/20
%Dependent functions: calc_K1, calc_K2, calc_CO2_loss_temp
%Inputs: S, CO2sat, kLa, pH, alk
%Outputs:CO2 loss to atmosphere without algae growth

%delete all figures and variables in the workspace
clear 
close all


%define variables
T = 20 + 273.15; %temp in Kelvins
sin = 25;
send = 45;
dels = 10;
PCO2 = 0.000416;
Tc = 20;
P = 10; %(dbar)
t = Tc*1.00024;
p = P/10;
kLa = 0.5; %(1/hr)
pHin = 6.5; %no units
pHend = 8.5; %no units
delpH = 0.1; %no units
d = .15; %(m) depth of pond
alk = 2.5;

%define r_temp
r_sal = calc_CO2_loss_sal (T, PCO2, t, p, alk, kLa, pHin, pHend, delpH, send, sin, dels);

%modify r for plotting
%x axis is the first column of r
x = r_sal(:,1);
r_sal(:,1) = [];
r = r_sal*d; %(g/m2*day)
L = {'-', '--', '-.'};

%plot CO2 loss vs pH for multiple alkalinities
hold on
plot(x,r(:,1), 'Linestyle', L{1})
plot(x,r(:,2), 'Linestyle', L{2})
plot(x,r(:,3), 'Linestyle', L{3})

xlabel('pH')
ylabel('CO_2 loss to the atmosphere (g m^{-2} day^{-1})')
legend('S = 25 g/kg','S = 35 g/kg','S = 45 g/kg')